### ISSUE TITLE
Add logger to KCP machine filters

### ISSUE THREAD

**Issue by sedefsavas** [kubernetes-sigs/cluster-api#3248]
Machine filters (controlplane/kubeadm/internal/macine_filters) do not return any errors hence adding a logger to this package will give some visibility into if/when an error occurs.

/help
/priority backlog

cc @detiber

**Comment by fabriziopandini** [comment]
I'm debated on this one.
I think we are pushing too much application logic, a most specifically code reading from the cluster, into machine filters.
So IMO a better approach to solve this problem is to have a unique method that collects all the required info upfront, with proper error management, and to use filters only ... as a filter.

**Comment by benmoss** [comment]
I agree with @fabriziopandini, the filters should just be doing filtering, so they should only take values that have sufficient information for doing the filtering, not stuff like `client.Client`.

**Comment by fabriziopandini** [comment]
This is not necessary after the refactor in https://github.com/kubernetes-sigs/cluster-api/pull/3405/files that moved the code that can generate errors to the `NewControlPlane` func
